### PR TITLE
[FSSDK-8837] Implement API for sendOdpEvent in Agent

### PR DESCRIPTION
**Pull request by Mat001**
## Summary
- Implement API for sendOdpEvent, update open api schema. Extend agent config to include odp.
- extend config to accommodate ODP
- update open API schema
- unit tests

## Issues
- [FSSDK-8837 Implement API for sendOdpEvent in Agent](https://jira.sso.episerver.net/browse/FSSDK-8837)

### CODE REVIEW

File: pkg/handlers/send_odp_event.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	}
+
+	body, err := getResponseBody(r)
+	if err != nil {
+		RenderError(err, http.StatusBadRequest, w, r)
+		return
+	}
+
+	success := optlyClient.SendOdpEvent(body.Action, body.Type, body.Identifiers, body.Data)
+	if err != nil {
```
Comment (jaeopt) [inline pkg/handlers/send_odp_event.go:46]: I think you mean `if !success`. This `err` is not the result of `SendOdpEvent`.

File: pkg/handlers/send_odp_event.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+		RenderError(err, http.StatusInternalServerError, w, r)
+		return
+	}
+
+	returnResult := optimizely.SendOdpEvent{
```
Comment (jaeopt) [inline pkg/handlers/send_odp_event.go:51]: I figured out this is for response formatting. The same name `SendOdpEvent` is confusing with the api call of the same name.

**Review by jaeopt — CHANGES_REQUESTED** [review]
Looks good. A few changes suggested.

File: api/openapi-spec/openapi.yaml [hunk 3]
```diff
@@ -551,6 +573,20 @@ components:
...
+      properties:
+        action:
+          type: string
+        type:
+          type: string
+        identifiers:
+          type: object
+          additionalProperties: true
+        data:
+          type: object
```
[the remark this answers is not in the document]
Comment (Mat001) [reply api/openapi-spec/openapi.yaml:586]: Thx @jaeopt . One thing: type: [string, integer, number, boolean] works with open api schema version 3.1
Whereas in our current schema I see we use version 3.0 syntax. So I propose this equivalnt for 3.0 version syntax.

identifiers:
          type: object
          additionalProperties:
            type: string
        data:
          type: object
          additionalProperties:
            oneOf:
              - type: string
              - type: integer
              - type: number
              - type: boolean

File: pkg/handlers/send_odp_event.go (hunk 2 above)
Comment (Mat001) [reply pkg/handlers/send_odp_event.go:51]: Yeah. Renaming to SendOdpEventResponseModel

File: pkg/handlers/send_odp_event.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+		return
+	}
+
+	body, err := getResponseBody(r)
+	if err != nil {
+		RenderError(err, http.StatusBadRequest, w, r)
+		return
+	}
+
+	success := optlyClient.SendOdpEvent(body.Action, body.Type, body.Identifiers, body.Data)
```
[the remark this answers is not in the document]
Comment (yasirfolio3) [reply pkg/handlers/send_odp_event.go:45]: Hey, so i looked into tis and we might have to make changes in eventManager as well as many other files to cater this. Currently `SendOdpEvent` returns the value it receives from `ProcessEvent` method of `EventManager` but yes its achievable.

[the remark this answers is not in the document]
Comment (jaeopt) [reply pkg/handlers/send_odp_event.go:45]: @yasirfolio3 Great! Can you create a ticket for the change? We can wrap up this PR with the current boolean and then later we can fix go and agent together.

[the remark this answers is not in the document]
Comment (jaeopt) [reply pkg/handlers/send_odp_event.go:45]: @Mat001

File: pkg/optimizely/optimizelytest/client.go [hunk 5]
```diff
@@ -18,13 +18,16 @@
...
+	"time"
+
 	"github.com/optimizely/go-sdk/pkg/client"
 	"github.com/optimizely/go-sdk/pkg/config"
 	"github.com/optimizely/go-sdk/pkg/decision"
 	"github.com/optimizely/go-sdk/pkg/entities"
 	"github.com/optimizely/go-sdk/pkg/event"
 	"github.com/optimizely/go-sdk/pkg/logging"
 	"github.com/optimizely/go-sdk/pkg/odp"
+	odp_event "github.com/optimizely/go-sdk/pkg/odp/event"
```
Comment (yasirfolio3) [inline pkg/optimizely/optimizelytest/client.go:30]:
```suggestion
	pkgOdpEvent "github.com/optimizely/go-sdk/pkg/odp/event"
```

**Review by yasirfolio3 — APPROVED** [review]
1 minor change suggested, looks good overall.

File: pkg/handlers/send_odp_event.go [hunk 6]
```diff
@@ -36,26 +36,26 @@ func SendOdpEvent(w http.ResponseWriter, r *http.Request) {
...
+	body, err := getRequestOdpEvent(r)
 	if err != nil {
 		RenderError(err, http.StatusBadRequest, w, r)
 		return
 	}
 
 	success := optlyClient.SendOdpEvent(body.Action, body.Type, body.Identifiers, body.Data)
-	if err != nil {
+	if !success {
 		RenderError(err, http.StatusInternalServerError, w, r)
```
Comment (jaeopt) [inline pkg/handlers/send_odp_event.go:47]: This "err" should be fixed later when go-sdk returns proper error for `SendOdpEvent`. Let's keep it for now.

**Review by jaeopt — APPROVED** [review]
LGTM

**Pull request merged**